src/utils/astChecking.ts

```diff
@@ -669,6 +670,8 @@ export function checkSane(unit: SourceUnit, ctx: ASTContext): void {
       checkDirectChildren(node, 'vSubExpression');
     } else if (node instanceof CairoAssert) {
       checkDirectChildren(node, 'vExpression');
+    } else if (node instanceof CairoTempVar) {
+      // TODO make sure vars are live after and declared before
```

Comment:
What about this TODO?

Reply:
This was left as a todo for a sanity check. It should check that the defined tempvar is indeed live. However, this node will never actually be hit because it should only introduced by this pass. I would be happy to have it removed.